In [1]:
import numpy as np
import pandas as pd

### Acquire and summarize the training dataset

In [2]:
# Acquire the training dataset

df = pd.read_csv("train_features.csv")
df.shape

(23814, 876)

In [3]:
# Take a look at the columns

df.head()

sig_id cp_type  cp_time cp_dose     g-0     g-1     g-2     g-3  \
0  id_000644bb2  trt_cp       24      D1  1.0620  0.5577 -0.2479 -0.6208   
1  id_000779bfc  trt_cp       72      D1  0.0743  0.4087  0.2991  0.0604   
2  id_000a6266a  trt_cp       48      D1  0.6280  0.5817  1.5540 -0.0764   
3  id_0015fd391  trt_cp       48      D1 -0.5138 -0.2491 -0.2656  0.5288   
4  id_001626bd3  trt_cp       72      D2 -0.3254 -0.4009  0.9700  0.6919   

      g-4     g-5  ...    c-90    c-91    c-92    c-93    c-94    c-95  \
0 -0.1944 -1.0120  ...  0.2862  0.2584  0.8076  0.5523 -0.1912  0.6584   
1  1.0190  0.5207  ... -0.4265  0.7543  0.4708  0.0230  0.2957  0.4899   
2 -0.0323  1.2390  ... -0.7250 -0.6297  0.6103  0.0223 -1.3240 -0.3174   
3  4.0620 -0.8095  ... -2.0990 -0.6441 -5.6300 -1.3780 -0.8632 -1.2880   
4  1.4180 -0.8244  ...  0.0042  0.0048  0.6670  1.0690  0.5523 -0.3031   

     c-96    c-97    c-98    c-99  
0 -0.3981  0.2139  0.3801  0.4176  
1  0.1522  0.1241  0.6077  0.7371  
2 -0.6417 -0.2187 -1.4080  0.6931  
3 -1.6210 -0.8784 -0.3876 -0.8154  
4  0.1094  0.2885 -0.3786  0.7125  

[5 rows x 876 columns]

In [4]:
df.sig_id.value_counts().size

23814

In [5]:
df.cp_type.value_counts()

trt_cp         21948
ctl_vehicle     1866
Name: cp_type, dtype: int64

In [6]:
df.cp_time.value_counts()

48    8250
72    7792
24    7772
Name: cp_time, dtype: int64

In [7]:
df.cp_dose.value_counts()

D1    12147
D2    11667
Name: cp_dose, dtype: int64

**Takeaways**
1. The trainning dataset has **23814 rows** and **876 columns**.
2. Cols: 
    - sig_id: MoA response(s) of 23814 samples
    - g- signify gene expression data
    - c- signify cell viability data
    - cp_type: indicates samples treated with a compound or with a control perturbation
        - trt_cp: 21948
        - ctl_vehicle: 1866
    - cp_time indicates treatment duration:
        - 24: 7772
        - 48: 8250
        - 72: 7792
    - cp_does indicates treatment dose:
        - D1: 12147
        - D2: 11667   
3. I need to figure out what g-number and c-number means.

In [18]:
# Create a mask for cell viability cols only

mask_viability = df.columns.str.contains('c-')

df_cell_viability = df.iloc[:, mask_viability]
df_cell_viability.shape

(23814, 100)

In [20]:
df_cell_viability.describe()

c-0           c-1           c-2           c-3           c-4  \
count  23814.000000  23814.000000  23814.000000  23814.000000  23814.000000   
mean      -0.355156     -0.455359     -0.480542     -0.377666     -0.493552   
std        1.752565      2.004962      2.039346      1.554583      2.091888   
min      -10.000000    -10.000000    -10.000000    -10.000000    -10.000000   
25%       -0.549300     -0.538425     -0.550375     -0.601975     -0.559975   
50%       -0.009000      0.009800     -0.004000     -0.039050      0.011400   
75%        0.448975      0.476875      0.460575      0.413100      0.470300   
max        3.365000      3.915000      2.970000      2.408000      4.555000   

                c-5           c-6           c-7           c-8           c-9  \
count  23814.000000  23814.000000  23814.000000  23814.000000  23814.000000   
mean      -0.446037     -0.574855     -0.266293     -0.451307     -0.458664   
std        1.893484      2.234152      1.465313      1.948829      1.646122   
min      -10.000000    -10.000000    -10.000000    -10.000000    -10.000000   
25%       -0.578775     -0.571250     -0.524275     -0.551100     -0.654675   
50%       -0.008900      0.007400      0.006350     -0.005350     -0.025350   
75%        0.442675      0.459475      0.469375      0.453175      0.424400   
max        3.614000      3.904000      4.918000      3.648000      3.194000   

       ...          c-90          c-91          c-92          c-93  \
count  ...  23814.000000  23814.000000  23814.000000  23814.000000   
mean   ...     -0.469244     -0.461411     -0.513256     -0.500142   
std    ...      2.000488      2.042475      2.001714      2.107105   
min    ...    -10.000000    -10.000000    -10.000000    -10.000000   
25%    ...     -0.566175     -0.565975     -0.589975     -0.568700   
50%    ...     -0.009900      0.003250     -0.009100     -0.013750   
75%    ...      0.457750      0.461500      0.445675      0.452900   
max    ...      4.069000      3.960000      3.927000      3.596000   

               c-94          c-95          c-96          c-97          c-98  \
count  23814.000000  23814.000000  23814.000000  23814.000000  23814.000000   
mean      -0.507093     -0.353726     -0.463485     -0.378241     -0.470252   
std        2.159589      1.629291      2.059725      1.703615      1.834828   
min      -10.000000    -10.000000    -10.000000    -10.000000    -10.000000   
25%       -0.563775     -0.567975     -0.552575     -0.561000     -0.592600   
50%       -0.003300     -0.010250     -0.001250     -0.006800      0.014000   
75%        0.470900      0.444750      0.465225      0.446400      0.461275   
max        3.747000      2.814000      3.505000      2.924000      3.111000   

               c-99  
count  23814.000000  
mean      -0.301505  
std        1.407918  
min      -10.000000  
25%       -0.562900  
50%       -0.019500  
75%        0.438650  
max        3.805000  

[8 rows x 100 columns]

**Takeaways**
- The cell viability has 100 columns.
- The min is -10.0 for all the columns. **I thought the min is 0. why negative number?** 
- The max is different among the columns. **Do I need to sacle the data?**